### PR TITLE
Add recovery service for compose-apps-early-start

### PR DESCRIPTION
**Pull request by ricardosalveti**

### CODE REVIEW

**Review by doanac — APPROVED** [review]
i think this should help. i'll keep poking some at dockerd, but we'll always need this as a last-result fail-safe.

File: meta-lmp-base/recipes-support/compose-apps-early-start/compose-apps-early-start/compose-apps-early-start [hunk 1]
```diff
@@ -1,5 +1,22 @@
 #!/bin/sh -e
 
+function start_compose_app() {
+	app=$1
+	cd /var/sota/compose-apps/${app}
+
+	# Temporarily disable e to catch possible issues with the initial up
+	set +e
```
Comment (MrCry0) [inline meta-lmp-base/recipes-support/compose-apps-early-start/compose-apps-early-start/compose-apps-early-start:8]: I'd recommend using the following trick to handle RC under `set -e`

```
echo "Starting app ${app}"
ret=0
docker-compose up -d || ret=$?
```
This method doesn't need to dis/en e

**Review by mike-sul — COMMENTED** [review]
LGTM.

Is the `compose-apps-early-start` is executed at each boot? If so, shouldn't it be deactivated once it's run once, so `dockerd` starts containers by itself at each boot after that?

**Review by angolini — APPROVED** [review]

**Comment by ricardosalveti** [comment]
> Is the `compose-apps-early-start` is executed at each boot? If so, shouldn't it be deactivated once it's run once, so `dockerd` starts containers by itself at each boot after that?

In this case it is actually a good thing it gets executed at each boot, because otherwise we would not be able to restore the system to a working stage.

And it is not a problem since docker will see that the container is already up and move forward with the boot.

File: meta-lmp-base/recipes-support/compose-apps-early-start/compose-apps-early-start/compose-apps-early-start (hunk 1 above)
Comment (ricardosalveti) [reply meta-lmp-base/recipes-support/compose-apps-early-start/compose-apps-early-start/compose-apps-early-start:8]: Nice, I knew there was a better way but forgot when doing this.

**Review by MrCry0 — APPROVED** [review]
LGTM

**Pull request merged**